src/osemgrep/networking/Unit_Login.ml

```diff
@@ -0,0 +1,151 @@
+(* Austin Theriault
+ *
+ * Copyright (C) 2019-2023 Semgrep, Inc.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * version 2.1 as published by the Free Software Foundation, with the
+ * special exception on linking described in file LICENSE.
+ *
+ * This library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the file
+ * LICENSE for more details.
+ *)
+
+(*****************************************************************************)
+(* Prelude *)
+(*****************************************************************************)
+open Testutil
+
+(*****************************************************************************)
+(* Helpers *)
+(*****************************************************************************)
+
+let ok_token = "ok_token"
+let bad_token = "bad_token"
+
+let login_session =
+  ( Uuidm.of_string "00000000-0000-0000-0000-000000000000" |> Option.get,
+    snd (Semgrep_login.make_login_url ()) )
+
+let fake_settings =
+  "has_shown_metrics_notification: true\n\
+   anonymous_user_id: 5a06eb03-071d-4c89-b297-993c4912fa8f"
+
+let with_login_client =
+  let make_fn req body =
+    let uri = Cohttp.Request.uri req in
+    match Uri.path uri with
+    | "/api/agent/deployments/current" ->
+        let status, body =
+          match Testing_client.get_header req "Authorization" with
+          | Some "Bearer ok_token" -> (200, "./tests/login/ok_response.json")
+          | Some "Bearer bad_token" -> (401, "./tests/login/bad_response.json")
+          | _ -> failwith "Unexpected token"
+        in
+        Lwt.return Testing_client.(basic_response ~status body)
+    | "/api/agent/tokens/requests" ->
+        let%lwt () =
+          Testing_client.check_body body "./tests/login/fetch_body.json"
+        in
+        Lwt.return
+          (Testing_client.basic_response ~status:200
+             "./tests/login/token_response.json")
+    | _ -> failwith ("Unexpected path: " ^ Uri.path uri)
+  in
+  Testing_client.with_testing_client make_fn
+
+let with_four_o_four_client =
+  let make_fn req body =
+    ignore body;
+    ignore req;
+    Lwt.return Testing_client.(basic_response ~status:404 "/dev/null")
+  in
+  Testing_client.with_testing_client make_fn
+
+let with_tmp_settings_file f () =
+  Common2.with_tmp_file ~str:fake_settings ~ext:"yml" (fun tmp_settings_file ->
+      let old_settings = !Semgrep_envvars.v in
```

Comment:
maybe don't call it settings? It's confusing as it's actually the content of the whole envvars.
Maybe old_env?

Reply:
how about with_mock_envvars?